### PR TITLE
chore: migrate $ERROR -> throw new Test262Error in test/built-ins/isFinite,isNaN

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: test/built-ins/isNaN/S15.1.2.4_A2.7.js [hunk 1]
```diff
@@ -13,9 +13,9 @@ description: >
 
 try {
   new isNaN();
-  $ERROR('#1.1: new isNaN() throw TypeError. Actual: ' + (new isNaN()));
+  throw new Test262Error('#1.1: new isNaN() throw TypeError. Actual: ' + (new isNaN()));
```
Comment (leobalter) [inline test/built-ins/isNaN/S15.1.2.4_A2.7.js:16]:
```suggestion
```

**Review by leobalter — CHANGES_REQUESTED** [review]

**Comment by leobalter** [comment]
these are not your fault, but we have a chance to improve this.

File: test/built-ins/isFinite/S15.1.2.5_A2.7.js [hunk 2]
```diff
@@ -13,9 +13,9 @@ description: >
 
 try {
   new isFinite();
-  $ERROR('#1.1: new isFinite() throw TypeError. Actual: ' + (new isFinite()));
+  throw new Test262Error('#1.1: new isFinite() throw TypeError. Actual: ' + (new isFinite()));
```
[the remark this answers is not in the document]
Comment (rwaldron) [reply test/built-ins/isFinite/S15.1.2.5_A2.7.js:16]: I'm not changing any test contents in this or any of the `$ERROR` migration PRs

**Comment by rwaldron** [comment]
> these are not your fault, but we have a chance to improve this.

We can make corrections in other PRs

**Comment by rwaldron** [comment]
I accepted the changes against my will :P

**Review by leobalter — APPROVED** [review]

**Comment by leobalter** [comment]
Thanks! The only issue is the certainty of never prioritizing these again after the PR.

**Pull request merged**